checksit/generic.py

```diff
@@ -251,3 +252,67 @@ def check_var(dct, variable, defined_attrs, skip_spellcheck=False):
                     )
 
     return errors, warnings
+
+
+def check_file_name(file_name, vocab_checks=None, **kwargs):
+    """
+    Checks format of file name
+
+    Works for NCAS-GENERAL, would work for NCAS-RADAR if radar scan type is added as data product
+    """
+    vocab_checks = vocab_checks or {}
+    errors = []
+    warnings = []
+    file_name_parts = file_name.split("_")
+
+    # check instrument name
+    if "instrument" in vocab_checks.keys():
+        if vocabs.check(vocab_checks["instrument"], file_name_parts[0], label="_") != []:
+            errors.append(f"[file name]: Invalid file name format - unknown instrument {file_name_parts[0]}")
+    else:
+        msg = "No instrument vocab defined in specs"
+        raise KeyError(msg)
+
+    # check platform
+    if "platform" in vocab_checks.keys():
+        if vocabs.check(vocab_checks["platform"], file_name_parts[1], label="_") != []:
+            errors.append(f"[file name]: Invalid file name format - unknown platform {file_name_parts[1]}")
+    else:
+        msg = "No platform vocab defined in specs"
+        raise KeyError(msg)
+    
+    # check date format
+    # could be yyyy, yyyymm, yyyymmdd, yyyymmdd-HH, yyyymmdd-HHMM, yyyymmdd-HHMMSS
+    # first checks format, then date validity 
+    if not re.match("^\d{4}$|^\d{6}$|^\d{8}$|^\d{8}-\d{2}$|^\d{8}-\d{4}$|^\d{8}-\d{6}$", file_name_parts[2]):
+        errors.append(f"[file name]: Invalid file name format - bad date format {file_name_parts[2]}")
+    else:
+        fmts = ("%Y", "%Y%m", "%Y%m%d", "%Y%m%d-%H", "%Y%m%d-%H%M", "%Y%m%d-%H%M%S")
+        valid_date_found = False
+        for f in fmts:
+            try:
+                t = dt.datetime.strptime(file_name_parts[2], f)
+                valid_date_found = True
+                break
+            except ValueError:
+                pass 
+        if not valid_date_found:
+            errors.append(f"[file name]: Invalid file name format - invalid date in file name {file_name_parts[2]}")
+
+    # check data product
+    if "data_product" in vocab_checks.keys():
+        if vocabs.check(vocab_checks["data_product"], file_name_parts[3], label="_") != []:
+            errors.append(f"[file name]: Invalid file name format - unknown data product {file_name_parts[3]}")
+    else:
+        msg = "No data product vocab defined in specs"
+        raise KeyError(msg)
+
+    # check version number format
```

Comment:
Another tiny issue. Since this piece of code gets called on consecutive lines (`file_name_parts[-1].split(".nc")[0]`), I would define a temporary variable:
```
version_component = file_name_parts[-1].split(".nc")[0]
```
And use that.

Reply:
@joshua-hampton: I've made 2 tiny suggestions for changes. Apart from that, it all looks good to me. Obviously, unit tests underpinning the file name checker would be ideal but I trust your judgement on that :-)